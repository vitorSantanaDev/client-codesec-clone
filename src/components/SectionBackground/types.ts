import { ReactNode } from 'react'

export type IViewProps = {
  children: ReactNode
  backgroundImagePath: string
}
