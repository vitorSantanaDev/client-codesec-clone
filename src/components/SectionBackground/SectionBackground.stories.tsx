import { ComponentStory, ComponentMeta } from '@storybook/react'
import SectionBackground from '.'

export default {
  title: 'SectionBackground',
  component: SectionBackground,
  args: {
    backgroundImagePath: '/images/bg-hero.png'
  }
} as ComponentMeta<typeof SectionBackground>

export const Default: ComponentStory<typeof SectionBackground> = (args) => (
  <SectionBackground {...args} />
)
