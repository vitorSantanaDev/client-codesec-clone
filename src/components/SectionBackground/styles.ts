import styled, { css } from 'styled-components'

type WrapperProps = {
  bgPath: string
}

export const Wrapper = styled.section<WrapperProps>`
  ${({ bgPath }) => css`
    width: 100%;

    background-size: cover;
    background-position: center;
    background-repeat: no-repeat;
    background-image: url(${bgPath});
  `}
`
