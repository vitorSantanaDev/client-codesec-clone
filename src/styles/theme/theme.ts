export const theme = {
  colors: {
    redPrimary: '#E83752',
    redLight: '#FA5770',
    redDark: '#D4203B',
    redError: '#EB3D3D',
    redGradient: 'linear-gradient(243.46deg, #FA5770 16.66%, #E83752 83.34%)',
    redGradientDark:
      'linear-gradient(225.05deg, #FA5770 0%, rgba(232, 55, 82, 0) 100%)',
    redGradientLight:
      'linear-gradient(225.05deg, rgba(250, 87, 112, 0.1) 0%, rgba(232, 55, 82, 0) 50%)',
    greenSucsses: '#32D957',
    white: '#FFFFFF',
    whiteLight: '#F5F0F1',
    whiteDark: '#EDE4E5',
    black: '#121111',
    blackLight: '#1C1B1B',
    blackDark: '#0A0909',
    gray1: '#141414',
    gray2: '#2C2C2C',
    gray3: '#444444',
    gray4: '#5C5C5C',
    gray5: '#747474',
    gray6: '#8C8C8C',
    gray7: '#A3A3A3'
  },
  font: {
    family: {
      default: "'Raleway', sans-serif"
    },
    sizes: {
      small: '1.6rem',
      medium: '2.4rem',
      large: '3.2rem',
      xlarge: '4.0rem',
      xxlarge: '4.8rem',
      huge: '5.6rem',
      xhuge: '6.4rem',
      xxhuge: '8rem'
    }
  },
  media: {
    lteMedium: '(max-width: 768px)',
    lteLarge: '(max-width: 820px)',
    ltexLarge: '(max-width: 912px)',
    lteSmall: '(max-width: 420px)',
    lteXSmall: '(max-width: 280px)'
  },
  spacings: {
    small: '1.6rem',
    medium: '2.4rem',
    large: '3.2rem',
    xlarge: '4.0rem',
    xxlarge: '4.8rem',
    huge: '5.6rem',
    xhuge: '6.4rem',
    xxhuge: '8rem'
  }
}
