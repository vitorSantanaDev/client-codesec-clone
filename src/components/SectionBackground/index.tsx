import { IViewProps } from './types'
import * as S from './styles'

const SectionBackground = ({ children, backgroundImagePath }: IViewProps) => (
  <S.Wrapper bgPath={backgroundImagePath}>{children}</S.Wrapper>
)

export default SectionBackground
