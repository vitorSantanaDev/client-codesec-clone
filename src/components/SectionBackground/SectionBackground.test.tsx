import { screen } from '@testing-library/react'
import { renderTheme } from 'styles/theme/render-theme'
import SectionBackground from '.'

describe('<SectionBackground />', () => {
  it('should render the section with background and heading', () => {
    const pathImage = '/images/bg-hero.png'
    const { container } = renderTheme(
      <SectionBackground backgroundImagePath={pathImage}>
        <h1>React App</h1>
      </SectionBackground>
    )
    const heading = screen.getByRole('heading', { name: /react app/i })
    expect(heading).toBeInTheDocument()
    expect(container.firstChild).toHaveStyle({
      width: '100%',
      'min-height': '100vh',

      'background-size': 'cover',
      'background-position': 'center',
      'background-repeat': 'no-repeat',
      'background-image': `url(${pathImage})`
    })
    expect(container.firstChild).toMatchSnapshot()
  })
})
