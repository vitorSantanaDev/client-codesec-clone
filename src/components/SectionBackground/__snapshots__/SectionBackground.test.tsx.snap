// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SectionBackground /> should render the section with background and heading 1`] = `
.c0 {
  width: 100%;
  background-size: cover;
  background-position: center;
  background-repeat: no-repeat;
  background-image: url(/images/bg-hero.png);
}

<section
  class="c0"
>
  <h1>
    React App
  </h1>
</section>
`;
